src/main/java/uk/gov/hmcts/reform/sscs/robotics/json/RoboticsJsonMapper.java

```
@@ -113,7 +122,14 @@ private static JSONObject buildAppealDetails(JSONObject obj, SscsCaseData sscsCa
         }
 
         if (appeal.getMrnDetails().getDwpIssuingOffice() != null) {
-            obj.put("pipNumber", appeal.getMrnDetails().getDwpIssuingOffice());
+            Optional<OfficeMapping> officeMapping = dwpAddressLookupService.getDwpMappingByOffice(appeal.getBenefitType().getCode(),
+                    appeal.getMrnDetails().getDwpIssuingOffice());
+            if (officeMapping.isPresent()) {
+                obj.put("pipNumber", officeMapping.get().getMapping().getGaps());
+            } else {
+                throw new DwpAddressLookupException(String.format("could not find dwp officeAddress for benefitType %s and dwpIssuingOffice %s",
```

are we sure we want to throw an exception here?